specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json

```diff
@@ -0,0 +1,2964 @@
+          "description": "Indicates whether the resource name is available."
+        },
+        "reason": {
+          "type": "string",
+          "description": "Reason for name being unavailable."
+        }
+      },
+      "description": "Represents a resource name availability."
+    },
+    "Identity": {
+      "x-ms-client-flatten": true,
+      "type": "object",
+      "properties": {
+        "principalId": {
+          "type": "string",
+          "description": "ObjectId from the KeyVault",
+          "readOnly": true
+        },
+        "tenantId": {
+          "type": "string",
+          "description": "TenantId from the KeyVault",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "description": "Type of managed service identity.",
+          "enum": [
+            "UserAssigned"
+          ],
+          "x-ms-enum": {
+            "name": "ManagedServiceIdentityType",
+            "modelAsString": false
+          }
+        },
+        "userAssignedIdentities": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "object",
+            "description": "",
+            "items": {
+              "$ref": "#/definitions/UserAssignedIdentity"
+            }
+          },
+          "description": "Metadata of user assigned identity."
+        }
+      },
+      "description": "Properties to configure Identity for Bring your Own Keys"
+    },
+    "UserAssignedIdentity": {
+      "type": "object",
+      "properties": {
+        "principalId": {
+          "description": "Principal Id of user assigned identity",
+          "type": "string",
+          "x-ms-client-name": "PrincipalId",
+          "readOnly": true
+        },
+        "clientId": {
+          "description": "Client Id of user assigned identity",
+          "type": "string",
+          "x-ms-client-name": "ClientId",
+          "readOnly": true
+        }
+      },
+      "description": "Metadata of user assigned identity."
+    },
+    "CutoverMigrationRequest": {
+      "type": "object",
+      "description": "Cutover migration request",
+      "properties": {
+        "dnsChangeMode": {
+          "type": "string",
+          "enum": [
+            "Update",
+            "Revert",
+            "None"
+          ],
+          "x-ms-enum": {
+            "name": "DnsChangeMode",
+            "modelAsString": true
+          },
+          "description": "The DNS Mode"
+        }
+      }
+    },
+    "MigrationAssessmentRequest": {
+      "type": "object",
+      "properties": {
+        "sourceServerResourceId": {
+          "type": "string",
+          "description": "Source Server Resource Id to verify and perform migration assessment on."
+        }
+      },
+      "description": "Request from client to perform migration assessment"
+    },
+    "MigrationAssessmentResource": {
+      "type": "object",
+      "properties": {
+        "sourceServerResourceId": {
```

Comment:
>sourceServerResourceId [](http://example.com/codeflow?start=9&length=22)

If you move the POST action to the resource, you wouldn't need this property in the response either. I see you are reusing this definition for replicas below, so removing this property might require some refactoring.

Reply:
> @pranavi-dandu - @NetRock asked me to also review your PR, so I've taken some time to look at it and provide comments, even though it's already approved by ARM.
> 
> Note, the first issue is that new API version PRs should be constructed using this process:
> 
> 1. Copy all files from current API version to new API version folder.
> 2. Make a base commit.
> 3. Apply changes needed for new API version in subsequent commits (including updating the API version in each file).
> 
> If you use OpenAPIHub to generate your PR, it will automatically use the correct format.
> 
> This really helps reviewers see what's being changed in the new API version. Most ARM reviewers will require you resubmit your new API version PR in the correct format.

Yes agree with you. I created this PR via openapihub only

Reply:
I will still need the SourceServerResourceID. It is not only used to represent primary but also the replica servers
Eg Response:
{
SourceServerResourceId : "......." (Primary source server resource Id )
ReplicaAssessmentReport : [
{
SourceServerResourceId: "......." (Replica source server resource Id 1)
}
{
SourceServerResourceId: "......." (Replica source server resource Id 2)
}
]
}

Reply:
OK, so the assessment of the source server also implicitly does assessment for all of the replicas of the source server. I found that part non-intuitive, since those are not really the source server (from the request). I think naming that property serverResourceId would actually be clearer.